Gemfile

```diff
@@ -43,3 +43,8 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+
+gem 'rack-cors'
+gem 'devise_token_auth'
```

Comment:
If you're planning on building a token-based authentication for the frontend, it's better to simply use `jwt` gem. It's much simpler. You just need of course to learn how to use it for both the back and frontend.